### ISSUE TITLE
Application Customizer unmounts after second lazy-load in new Lists UI

### ISSUE THREAD

**Issue by gerkevgarderen** [SharePoint/sp-dev-docs#10034]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [X] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [X] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- Microsoft Edge Version 131.0.2903.70, and Mozilla Firefox 133.0 
- SPFx version 1.19.0
- Node.js version 18

### Describe the bug / error

When you scroll down in a Sharepoint list, in the new UI (Lists experience), and it lazy-loads new items twice, the Application Customizer component disappears.

Only thing of note, when using the template Application Customizer component: I can see the `onDispose` is called for the Application Customizer component:
`[HelloWorldApplicationCustomizer._onDispose] Disposed custom top and bottom placeholders.`

Is anyone else seeing issues with Application Customizers disappearing when you scroll down far enough in a new UI style List?
We use this in combination with Command Customizers, so only seeing the new behaviour since the past few days.

### Steps to reproduce

1. Load spfx with an Application Customizer and a Command Customizer.
2. Open a list where the App customizer and Command customizer are both active
3. All is well so far. Application customizer is there, Command Customizer too.
4. Scroll down to the bottom, new rows load for the first time. All is still well.
5. Scroll down to bottom again, new rows load. Now the Application Customizer unloads. No console errors.

To be precise, the unmount / onDispose is called immediately _after_ loading and rendering new listitems.

### Expected behavior

The application customizer does not unload/umount when scrolling, or at least remounts after rendering the newly lazy-loaded items.

**Comment by joelfmrodrigues** [comment]
@gerkevgarderen I think I am seeing the same issue. Is you application customizer also disposed if you navigate between folders?

**Comment by joelfmrodrigues** [comment]
Created a sample solution to repro this

**Comment by gerkevgarderen** [comment]
Thanks for looking into this @joelfmrodrigues and good repro imo.

Interesting that for us it also happens for lists (instead of a Document Library), without needing to be in a (sub-)folder.
Relevant to note: for these sites we are limited to old SharePoint Sites, with the UI upgraded to Modern (and now the new Lists UI of course). Could be that this is a factor causing it to happen for non-document library type lists without folders.

Observed behaviour aligns perfectly with your video (without needing a subfolder): the Customizer does not crash, but unmounts/disposes, without clear cause.

**Comment by natalieethell** [comment]
@joelfmrodrigues @gerkevgarderen thank you for your report. Sorry for any disruptions in your work. Just rolled back a change that should fix this issue - can you confirm if you're seeing the customizers stay now?

**Comment by joelfmrodrigues** [comment]
@natalieethell I can't thank you enough! Just quickly tested on 3 different tenants and it seems to be resolved

**Comment by gerkevgarderen** [comment]
Yes, can confirm this is fixed now after force refresh. Thanks so much!
